KITE-Extras/src/main/java/io/cosmosoftware/kite/instrumentation/Scenario.java

```diff
@@ -8,147 +8,194 @@
-      commandList.put(gw, "");
-    }
+  public Scenario(JsonObject jsonObject, Logger logger, Networks networks) throws Exception {
+    this.logger = logger;
     String missingKey="";
-    String gateway;
-    String command;
+    this.networks = networks;
     try {
-      missingKey = "commandList";
-      JsonArray jsonArray = jsonObject.getJsonArray("commandList");
-      for (int j = 0 ; j < jsonArray.size() ; j++) {
-        JsonObject jsonObject2 = jsonArray.getJsonObject(j);
-        NWCommands nwCommands = new NWCommands(jsonObject2, this.instrumentation);
-        gateway = nwCommands.getGateway();
-        command = this.commandList.get(gateway);
-        this.commandList.remove(gateway);
-        if (!command.equals("")) {
-          command += "|| true && " + nwCommands.getCommand();
-        } else {
-          command += nwCommands.getCommand();
+      missingKey = "type";
+      this.type = jsonObject.getString("type");
+      if (this.type.equals("client")) {
+        missingKey = "clientIds";
+        JsonArray jsonArray = jsonObject.getJsonArray("clientIds");
+        for (int i = 0; i < jsonArray.size(); i++) {
+          this.clientIds.add(jsonArray.getInt(i));
         }
-        commandList.put(gateway, command);
+        this.gateway = null;
+      } else if (this.type.equals("gateway")) {
+        missingKey = "gateway";
+        this.gateway = jsonObject.getString("gateway");
+        this.clientIds = null;
+      } else {
+        throw new KiteTestException("The type specified doesn't exist", Status.FAILED);
       }
-    } catch (NullPointerException e) {
-      throw new KiteTestException("Error in json config scenario, the key " + missingKey + " is missing.", Status.FAILED, e);
+      missingKey = "network";
+      this.network = jsonObject.getString("network");
+      this.command =  this.networks.get(this.network).getProfile().getCommand();
+      missingKey = "name";
+      name = jsonObject.getString("name");
+    } catch (Exception e) {
+      throw new KiteTestException("The key " + missingKey + " is missing", Status.FAILED, e);
     }
-    name = jsonObject.getString("name", "Scenario number : " + i);
-    duration = jsonObject.containsKey("duration") ? jsonObject.getInt("duration") : 10000;
-    this.logger = logger;
   }
 
   public String getName() {
     return name;
   }
 
-  public int getClientId() {
-    return clientId;
+  public ArrayList<Integer> getClientIds() {
+    return clientIds;
+  }
+
+  public String getCommand() {
+    return command;
   }
 
-  public HashMap<String, String> getCommandList() {
-    return commandList;
+  public String getGateway() {
+    return gateway;
   }
 
-  public Integer getDuration() {
-    return duration;
+  public String getType() {
+    return type;
   }
 
-  public String runCommands() {
+  public String sendCommand(WebDriver webDriver, Instrumentation instrumentation, String remoteAddress, String GridId, String instrumentUrl) {
```

Comment:
too much repeated code. pls create reusable functions and use them in sendCommand and cleanUp